compiler/src/dotty/tools/dotc/typer/Applications.scala

```diff
@@ -2076,16 +2076,27 @@ trait Applications extends Compatibility {
   def resolveOverloaded(alts: List[TermRef], pt: Type)(using Context): List[TermRef] =
     record("resolveOverloaded")
 
-    /** Is `alt` a method or polytype whose result type after the first value parameter
+    /** Is `alt` a method or polytype whose approximated result type after the first value parameter
      *  section conforms to the expected type `resultType`? If `resultType`
      *  is a `IgnoredProto`, pick the underlying type instead.
+     *
+     *  Using an approximated result types is necessary to avoid false negatives
```

Comment:
Your grammar as right the first time, either "approximated result types" or "an approximated result type"